pkg/agent/baker.go

```diff
@@ -435,6 +435,28 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 			}
 			return 0
 		},
+		"ShouldConfigMemLock": func() bool {
+			return profile.CustomLinuxOSConfig != nil &&
```

Comment:
potentially add a helper on the top level profile for GetUlimitConfig which handles the nil checks?